src/v/cluster/health_monitor_backend.cc

```diff
@@ -826,4 +827,24 @@ bool health_monitor_backend::does_raft0_have_leader() {
     return _raft0->get_leader_id().has_value();
 }
 
+void health_monitor_backend::reset() {
+    // Explicitly reset or wipe state here instead of calling stop() because
+    // stop() will also close the gate and break the mutex
+
+    // Do not clear _node_callbacks because this is used to notify other
+    // structures, such as the feature manager, of changes in cluster health
+
+    vlog(clusterlog.info, "Resetting Health Monitor Backend");
+    _last_replies.clear();
+    _reports.clear();
+    _status.clear();
+    abort_current_refresh(errc::reset_in_progress);
+
+    if (_refresh_request) {
+        _refresh_request.release();
+    }
+    _last_refresh = ss::lowres_clock::time_point{};
+    _reports_disk_health = storage::disk_space_alert::ok;
```

Comment:
i get that we are resetting state, so maybe it doesn't matter. but i wonder if instead of resetting the state by clearing all the data structures, we should force a refresh? i mean, if we clear these structures then won't a refresh happen shortly after that? and if so, will we run the risk of causing something like the disk space alert to flap?

Reply:
I agree with this +1

Reply:
I put this change in a draft in #9480 to collect feedback